core/txpool/ingress_filters.go

```diff
@@ -0,0 +1,52 @@
+package txpool
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/types/interoptypes"
+)
+
+// IngressFilter is an interface that allows filtering of transactions before they are added to the transaction pool.
+// Implementations of this interface can be used to filter transactions based on various criteria.
+// FilterTx will return true if the transaction should be allowed, and false if it should be rejected.
+type IngressFilter interface {
+	FilterTx(tx *types.Transaction) bool
+}
+
+type interopFilter struct {
+	logsFn  func(tx *types.Transaction) ([]*types.Log, error)
+	checkFn func(ctx context.Context, ems []interoptypes.Message, safety interoptypes.SafetyLevel) error
+}
+
+func NewInteropFilter(
+	logsFn func(tx *types.Transaction) ([]*types.Log, error),
+	checkFn func(ctx context.Context, ems []interoptypes.Message, safety interoptypes.SafetyLevel) error) IngressFilter {
+	return &interopFilter{
+		logsFn:  logsFn,
+		checkFn: checkFn,
+	}
+}
+
+// FilterTx implements IngressFilter.FilterTx
+// it gets logs checks for message safety based on the function provided
+func (f *interopFilter) FilterTx(tx *types.Transaction) bool {
+	logs, err := f.logsFn(tx)
+	if err != nil {
+		return true // default to allow if logs cannot be retrieved
```

Comment:
Is this safe? Should we allow txs by default if there is an error in validating them?

Reply:
I think it's better to drop. The user can re-submit the tx if the tx was really valid and the issue was temporary. But we hurt block-building if a tx goes through that is not ready. At a later point we can introduce some queuing mechanism for the type of txs that will only be valid later.